### ISSUE TITLE
False positive `WRONG_OVERLOADING_FUNCTION_ARGUMENTS`

### ISSUE THREAD

**Issue by nulls** [saveourtool/diktat#1594]
## Describe the bug
False positive `WRONG_OVERLOADING_FUNCTION_ARGUMENTS` for overridden methods:
```kotlin
fun list(projectCoordinates: ProjectCoordinates): Flux<FileKey> = list()
    .filter { it.projectCoordinates == projectCoordinates }

override fun list(): Flux<FileKey> = newFileStorage.list()
    .map { fileDto ->
        FileKey(
            projectCoordinates = fileDto.projectCoordinates,
            name = fileDto.name,
            uploadedMillis = fileDto.uploadedTime.toInstant(TimeZone.UTC).toEpochMilliseconds()
        )
    }
```

## Expected behavior
Do not take into account overridden methods for rule `WRONG_OVERLOADING_FUNCTION_ARGUMENTS`

## Environment information
* diktat version: 1.2.4.1
* build tool (maven/gradle): gradle
* how is diktat run (CLI, plugin, etc.): plugin
* kotlin version: 1.7.20
* operating system: unix,windows
* link to a project (if your project is public):

## Related

 * #1526
 * #1593

**Comment by icemachined** [comment]
the issue is not reproduced in diktat version 1.2.4.2, 1.2.4.1 was cut before #1520 was merged
